packages/sdk-core/CHANGELOG.md

```diff
@@ -29,6 +31,19 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Internal
 - Use `eslint-plugin-prettier` over separate `prettier` instance
 
+### Breaking
+- The `SuperToken` class is now an abstract base class and no longer contains the functions `upgrade` and `downgrade`.
+- `underlyingToken` is possibly undefined on `SuperToken`: `WrapperSuperToken` has `underlyingToken`, but `PureSuperToken` and `NativeAssetSuperToken` do not.
+> NOTE: These changes are due to the split of `SuperToken` into `WrapperSuperToken`, `PureSuperToken` and `NativeAssetSuperToken` classes.
+  - Migration: if you are using TypeScript, you will need to specify the type of token you are attempting to load:
+  ```
+  import { NativeAssetSuperToken } from "@superfluid-finance/sdk-core";
+  ...
+  const nativeAssetSuperToken = await framework.loadSuperToken<NativeAssetSuperToken>("ETHx");
```

Comment:
I still dislike that this doesn't do any runtime check that it's actually of that type.

Reply:
pls elaborate

Reply:
This will not fail:
`const nativeAssetSuperToken = await framework.loadSuperToken<NativeAssetSuperToken>("USDCx");`

It should.

Reply:
what would the fix be then?

Reply:
We went with option B.